### PR TITLE
feature: #33 | Add DockerFile and feature: #34Rename the users_servic…

### PR DESCRIPTION
**Pull request by astha-singh-dev**
…e to users-service in the settings.gradle.kts and the group name in the build.gradle.kts

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ENV PG_USERNAME $PG_USERNAME
+ENV PG_PASSWORD $PG_PASSWORD
+ENV MONGO_CONNECTION_URI $MONGO_CONNECTION_URI
+ENV MONGO_DATABASE $MONGO_DATABASE
+ENV MAIL_USERNAME $MAIL_USERNAME
+ENV MAIL_PASSWORD $MAIL_PASSWORD
+ENV OTP_SECRET_TOKEN $OTP_SECRET_TOKEN
+ENV AUTH_TOKEN_PUBLIC_KEY $AUTH_TOKEN_PUBLIC_KEY
+ENV AUTH_TOKEN_PRIVATE_KEY $AUTH_TOKEN_PRIVATE_KEY
+RUN ./gradlew clean assemble
```
Comment (ShivamNagpal) [inline Dockerfile]: It should be `./gradlew clean build`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ARG MAIL_PASSWORD
+ARG OTP_SECRET_TOKEN
+ARG AUTH_TOKEN_PUBLIC_KEY
+ARG AUTH_TOKEN_PRIVATE_KEY
+WORKDIR /app
+COPY gradle gradle
+COPY src src
+COPY gradlew gradlew
+COPY build.gradle.kts build.gradle.kts
+COPY settings.gradle.kts settings.gradle.kts
```
Comment (ShivamNagpal) [inline Dockerfile]: Copy everything else except src and then put `RUN ./gradlew check --dry-run` and then copy src to allow caching of the Gradle dependencies

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ARG PG_DATABASE
+ARG PG_USERNAME
+ARG PG_PASSWORD
+ARG MONGO_CONNECTION_URI
+ARG MONGO_DATABASE
+ARG MAIL_USERNAME
+ARG MAIL_PASSWORD
+ARG OTP_SECRET_TOKEN
+ARG AUTH_TOKEN_PUBLIC_KEY
+ARG AUTH_TOKEN_PRIVATE_KEY
```
Comment (ShivamNagpal) [inline Dockerfile]: Remove all the args and env from the Builder phase, as they are not needed

**Review by ShivamNagpal — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (ShivamNagpal) [reply Dockerfile]: Once you do this, you will require to copy the spotless.xml as well, otherwise the build will fail because of file linking issues

Comment (astha-singh-dev) [reply Dockerfile]: fixed

**Review by ShivamNagpal — APPROVED** [review]

**Pull request merged**